cmd/clusterctl/pkg/client/cluster/mover_test.go

```diff
@@ -255,7 +255,7 @@ var moveTests = []struct {
 				"/v1, Kind=Secret, ns1/cluster1-ca",
 				"/v1, Kind=Secret, ns1/cluster1-kubeconfig",
 				"bootstrap.cluster.x-k8s.io/v1alpha3, Kind=DummyBootstrapConfigTemplate, ns1/mp1",
-				"cluster.x-k8s.io/v1alpha3, Kind=MachinePool, ns1/mp1",
+				"exp.cluster.x-k8s.io/v1alpha3, Kind=MachinePool, ns1/mp1",
```

Comment:
Should the feature flags also be extended to `clusterctl`? I'm just trying to think if there will be any gotchas to user expectations if we have handling around the experimental types here by default and without some user input or management cluster introspection.

Reply:
I think clusterctl is safe to assume feature gates are enabled; otherwise, you have to make sure you match what's set up in the controllers (which easily could have been set up by someone else)

Reply:
I'm just wondering if we'll encounter any issues (such as types not being enabled for experimental features) that would cause issues if we try to operate on experimental types.

Reply:
That should be fine, clusterctl doesn't assume anything regarding which API types are referenced from objects, it'll try to move references by building a tree.